src/Browser.php

```diff
@@ -335,4 +337,64 @@ public function withOptions(array $options)
 
         return $browser;
     }
+
+    public function withTimeout($timeout)
+    {
+        $browser = clone $this;
+        $browser->transaction = $this->transaction->withOptions(array(
+            'timeout' => $timeout,
+        ));
+
+        return $browser;
+    }
+
+    public function withoutTimeout()
```

Comment:
`withTimeout()` LGTM, but I'm unsure about `withoutTImeout()`.

This currently sets the underlying open to `null` which returns to the default timeout value (60) as per the documentation. It does not send requests "without" a timeout.

I'm undecided, do we need dedicated methods or should we accept different values like the underlying option?

Reply:
Only added it because you mentioned in our Gitter communication. Just dropped it in the latest push.